File: app/src/main/java/com/github/alexeygorovoy/moxytemplate/rx/AppRxSchedulers.java
package com.github.alexeygorovoy.moxytemplate.rx;

import rx.Scheduler;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;


public class AppRxSchedulers extends RxSchedulers {

    @Override
    public Scheduler mainThread() {
        return AndroidSchedulers.mainThread();
    }

    @Override
    public Scheduler io() {
        return Schedulers.io();
    }

    @Override
    public Scheduler computation() {
        return Schedulers.computation();
    }
}

File: app/src/main/java/com/github/alexeygorovoy/moxytemplate/rx/AppRxSchedulers.kt
package com.github.alexeygorovoy.moxytemplate.rx

import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers


class AppRxSchedulers: RxSchedulers() {

    override fun mainThread(): Scheduler {
        return AndroidSchedulers.mainThread()
    }

    override fun io(): Scheduler {
        return Schedulers.io()
    }

    override fun computation(): Scheduler {
        return Schedulers.computation()
    }
}
